core/Symbols.h

```diff
@@ -617,12 +617,31 @@ class ClassOrModule final {
         return members_;
     };
 
+    struct MembersStableOrder {
+        std::vector<std::pair<NameRef, SymbolRef>> members;
+
+        template <typename P> void accumulateMembers(const GlobalState &gs, const ClassOrModule &klass, P predicate) {
+            size_t start = members.size();
+
+            for (const auto &e : klass.members()) {
+                if (predicate(e.first, e.second)) {
+                    members.emplace_back(e);
+                }
+            }
+
+            sortMembersStableOrder(gs, members.begin() + start, members.end());
```

Comment:
Do we benefit much from having this method exposes the `state` directly, instead of making that an implementation detail that gets returned?

It seems like the change would get simpler if we just returned a vector and hid the internal state. We wouldn't need the range-based sort function, wouldn't need to have people allocate a `state` before calling the function, etc.

I guess it saves needing to allocate a vector for each ancestor? Is that such a big win, compared to the win of only sorting the things that actually match, and versus the complexity of having a more stateful API?

(Context: part of the reason I didn't review this PR over the weekend was that when I opened it on my phone it looked more complex than the title made me expect it would be.)